File: submodules/mo_grippers/examples/dh_test.py
import time
from mo_grippers.grippers import DhGripper

gripper = DhGripper(port="/dev/ttyUSB1")
gripper.initialize(should_wait=True)
gripper.start_thread(read_rate=100) # 100Hz

gripper.close_fingers()
for _ in range(10):
    print("Current position: ", gripper.current_position)
    time.sleep(0.1)

gripper.open_fingers()
for _ in range(10):
    print("Current position: ", gripper.current_position)
    time.sleep(0.1)

print("Closing read thread and client connection!!")
gripper.close()

File: submodules/mo_grippers/examples/dh_test.cpp
#include <chrono>
#include <iostream>
#include <thread>
#include <plaif/mo-grippers/grippers/dh-gripper.h>

using namespace plaif;

int main()
{
  DhGripper gripper("/dev/ttyUSB1");
  gripper.initialize(true);
  gripper.startThread(100);

  gripper.closeFingers();
  for (int i = 0; i < 10; i++)
  {
    std::cout << "Current position: " << gripper.getCurrentPosition() << std::endl;
    std::this_thread::sleep_for(std::chrono::milliseconds(100));
  }

  gripper.openFingers();
  for (int i = 0; i < 10; i++)
  {
    std::cout << "Current position: " << gripper.getCurrentPosition() << std::endl;
    std::this_thread::sleep_for(std::chrono::milliseconds(100));
  }

  std::cout << "Closing read thread and client connection!!" << std::endl;
  gripper.close();

  return 0;
}
